corehq/apps/api/resources/v0_4.py

```diff
@@ -73,15 +73,28 @@ def xform_es(self, domain):
         return MOCK_XFORM_ES or XFormES(domain)
 
     def obj_get_list(self, bundle, domain, **kwargs):
+        include_archived = 'include_archived' in bundle.request.GET
         try:
-            es_query = es_search(bundle.request, domain)
+            es_query = es_search(bundle.request, domain, ['include_archived'])
         except Http400 as e:
             raise BadRequest(e.message)
-        es_query['filter']['and'].append({'term': {'doc_type': 'xforminstance'}})
+        if include_archived:
+            es_query['filter']['and'].append({'or': [
+                {'term': {'doc_type': 'xforminstance'}},
+                {'term': {'doc_type': 'xformarchived'}},
+            ]})
+        else:
+            es_query['filter']['and'].append({'term': {'doc_type': 'xforminstance'}})
+
+        def wrapper(doc):
+            if doc['doc_type'] in doc_types():
+                return doc_types()[doc['doc_type']].wrap(doc)
```

Comment:
Not a big deal, but you could save a bunch of dict allocations if you assigned the result of `doc_types()` to a variable instead of calling it here. Maybe like this:
```python
def wrapper(doc, doc_types=doc_types()):
    # use doc_types instead of doc_types() in here
```